### ISSUE TITLE
[HOLD for 53674] [$250] Sage - synced connection only shows after page is refreshed 

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53548]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.71-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5296970&group_by=cases:section_id&group_id=319504&group_order=asc
**Email or phone of affected tester (no customers):** N/A
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Navigate to https://staging.new.expensify.com/
2. Log in with a new Gmail account
3. Click on FAB - New workspace
4. Enable "Accounting" in the "More features" page.
5. Navigate to "Accounting"
6. Connect to Sage Intacct and upgrade the workspace to Control when asked

## Expected Result:
Sync should finish and the integration should load after.

## Actual Result:
Page is stuck at "Waiting for imported data to load". The synced integration shows only after the page is refreshed.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/ba06815e-70ff-420c-a1c7-8e764926f48a

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866633870141061573</li>
        <li>Upwork Job ID: 1866633870141061573</li>
        <li>Last Price Increase: 2024-12-17</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @rayane-djouah</details>

**Comment by stephanieelliott** [comment]
I think this may have been related to the site slows we had been experiencing, it is loading now: 

<img width="606" alt="image" src="https://github.com/user-attachments/assets/e461aa43-b701-4c85-8172-7b8d06e64acc">

**Comment by lanitochka17** [comment]
Issue is still reproducible on the latest build 9.0.73-8

https://github.com/user-attachments/assets/a615cddb-c0c3-4804-977f-8731f39cab63

**Comment by stephanieelliott** [comment]
I was able to reproduce this on a test Gmail account. Here are some logs that may be helpful: 

 - Intacct request marshalled: 
[is_temp_logsec741500-1def-4990-ab13-2b03c2679d70.log](https://github.com/user-attachments/files/18174053/is_temp_logsec741500-1def-4990-ab13-2b03c2679d70.log)

 - Intacct response: 
[is_temp_logs60b18470-9725-4d1c-adfe-061f2e0bc997.log](https://github.com/user-attachments/files/18174060/is_temp_logs60b18470-9725-4d1c-adfe-061f2e0bc997.log)

The API response says the connection was successful, however it appears stuck at `Waiting for imported data to load`. But you refresh the page, you can see that Sage is connected: 

<img width="962" alt="image" src="https://github.com/user-attachments/assets/1e9377d6-ae2c-4ed7-8965-f703b060b009" />

----

So this is not so much a stuck sync issue, it's a more of an issue with the page displaying update. Edited the issue title and body to make that more clear

**Comment by stephanieelliott** [comment]
Oh actually let's hold on https://github.com/Expensify/App/issues/53674 -- seems like that may be universal fix for all integrations

**Comment by stephanieelliott** [comment]
Moving this to weekly while we are on hold.

**Comment by stephanieelliott** [comment]
This is fixed now! Data loads right away: 

<img width="1016" alt="Image" src="https://github.com/user-attachments/assets/978d4661-ca69-46d0-a3de-93a7164d595d" />
